plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ImplementAvgBigint.java

```diff
@@ -13,52 +13,14 @@
  */
 package io.trino.plugin.clickhouse;
 
-import io.trino.matching.Capture;
-import io.trino.matching.Captures;
-import io.trino.matching.Pattern;
-import io.trino.plugin.base.expression.AggregateFunctionRule;
-import io.trino.plugin.jdbc.JdbcColumnHandle;
-import io.trino.plugin.jdbc.JdbcExpression;
-import io.trino.plugin.jdbc.JdbcTypeHandle;
-import io.trino.spi.connector.AggregateFunction;
-import io.trino.spi.expression.Variable;
-
-import java.sql.Types;
-import java.util.Optional;
-
-import static com.google.common.base.Verify.verify;
-import static io.trino.matching.Capture.newCapture;
-import static io.trino.plugin.base.expression.AggregateFunctionPatterns.basicAggregation;
-import static io.trino.plugin.base.expression.AggregateFunctionPatterns.expressionType;
-import static io.trino.plugin.base.expression.AggregateFunctionPatterns.functionName;
-import static io.trino.plugin.base.expression.AggregateFunctionPatterns.singleInput;
-import static io.trino.plugin.base.expression.AggregateFunctionPatterns.variable;
-import static io.trino.spi.type.BigintType.BIGINT;
-import static io.trino.spi.type.DoubleType.DOUBLE;
-import static java.lang.String.format;
+import io.trino.plugin.jdbc.expression.BaseImplementAvgBigint;
 
 public class ImplementAvgBigint
-        implements AggregateFunctionRule<JdbcExpression>
+        extends BaseImplementAvgBigint
 {
-    private static final Capture<Variable> INPUT = newCapture();
-
-    @Override
-    public Pattern<AggregateFunction> getPattern()
-    {
-        return basicAggregation()
-                .with(functionName().equalTo("avg"))
-                .with(singleInput().matching(variable().with(expressionType().equalTo(BIGINT)).capturedAs(INPUT)));
-    }
-
     @Override
-    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    protected String pushedExpression(String columnName)
     {
-        Variable input = captures.get(INPUT);
-        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignment(input.getName());
-        verify(aggregateFunction.getOutputType() == DOUBLE);
-
-        return Optional.of(new JdbcExpression(
-                format("avg((%s * 1.0))", context.getIdentifierQuote().apply(columnHandle.getColumnName())),
-                new JdbcTypeHandle(Types.DOUBLE, Optional.of("double"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty())));
+        return String.format("avg((%s * 1.0))", columnName);
```

Comment:
Looking at pushedExpression method name i wouldn't know it's responsible for coercing the column to a double.

`castToDouble(String columnName)`

also, column name must be quoted inside SQL expression

Reply:
Sure I can change the name of the method.  

About the quoting of the column name inside the expression, I kept the behavior the same from before. But I can definitely add quoting 👍

Reply:
Yup confirmed the column is being quoted, I'll change the name of the method arg to reflect that:
`castToDouble(String quotedColumnName)`

Reply:
@findepi I'm not sure what you mean, are you saying to change the name of the argument to `expression` or are you saying refactor this implementation to avoid handling a `quotedColumnName` entity at all.

Reply:
i'll leave this up to @hashhar

Reply:
I think Piotr's suggestion makes sense. I had a similar concern in https://github.com/trinodb/trino/pull/9857#discussion_r743032380.

Maybe you can change the `getRewriteExpression` to just return a format-string and then `String.format(expression, column)` in the base class instead of subclasses. That way it's difficult for subclasses to write "bad" expressions - since the superclass can do things like adding proper parenthesis etc.

(Or an alternative refactor that leads to same goal - keep as much "intelligence" in the base class as possible).

Reply:
Take another look, changed the method to return a format string per this discussion. 

I decided to *not* move the `avg` bit into the base class for 2 reasons:
1. The `getRewriteFormatExpression` method being a whole expression that is pushed to the remote db is conceptually simpler than it being a partial expression that is processed in the base class.
2. Some databases may potentially have a different name for `avg`, a tiny bit of duplication seems fine to be more flexible.